### PR TITLE
[web] Remove unnecessary CSS styles on <p> elements

### PR DESCRIPTION
**Pull request by mdebbar**
With https://github.com/flutter/engine/pull/31907, we started to absolutely position all the spans inside the paragraph, so there's no layout/alignment happening by the DOM. This means we can remove all the CSS styles that were used to make the browser lay things out correctly.

- Rename `<p>` and `<span>` elements to `<flt-paragraph>` and `<flt-span>` respectively.
- Remove a whole bunch of CSS that we don't need anymore.
- Fix tests.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 5 new digest(s) on patchset 6.
View them at https://flutter-engine-gold.skia.org/cl/github/32043

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/web_ui/lib/src/engine/text/canvas_paragraph.dart [hunk 1]
```diff
@@ -175,7 +173,7 @@ class CanvasParagraph implements ui.Paragraph {
         final RangeBox box = boxes[j++];
 
         if (box is SpanBox) {
-          lastSpanElement = html.document.createElement('span') as html.HtmlElement;
+          lastSpanElement = html.document.createElement('flt-span') as html.HtmlElement;
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/text/canvas_paragraph.dart:176]: Custom elements behave like `div`s right? Do you need `display: inline` (or `inline-block`) to retain what was special of using a `span` tag earlier? Or are we just doing display:block on everything?

**Review by ditman — APPROVED** [review]
LGTM! Small question about the "new spans"

File: lib/web_ui/lib/src/engine/text/canvas_paragraph.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/text/canvas_paragraph.dart:176]: That's a good point.

The display property doesn't matter anymore. Each `<flt-span>` contains a single fragment of text that can't be broken into multiple lines.

A single line of text that is positioned absolutely shouldn't be affected by `display:block` or `display:inline`.

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 9.
View them at https://flutter-engine-gold.skia.org/cl/github/32043

**Pull request merged**